src/util/defs.h

```diff
@@ -3,26 +3,39 @@
 #include <Qt>
 
 // Maximum buffer length to each EngineObject::process call.
-//TODO: Replace this with mixxx::AudioParameters::bufferSize()
-constexpr unsigned int MAX_BUFFER_LEN = 160000;
+// Note: MAX_BUFFER_LEN shouldn't be use for audio processing,
+// because the buffers should be sized according to the processing
+// block in frames and then multiplied by the channel count.
+// Then audio buffers used in I/O and non realtime operation
+// should be sized according to their task with a balance in mind
+// between CPU efficiency and memory usage.
+//
+// Max frames at 44.1kHz:     4096
+// Max frames at  192kHz: 4 * 4096
+//
+// TODO: Replace this with mixxx::AudioParameters::bufferSize()
+static constexpr unsigned int kMaxProcessFrames = 4 * 4096;
```

Comment:
Where is the 4 coming from? 
Mixxx supports audio buffers up to 80 ms. This is 8192 Frames @ 192kHz

Reply:
The 4 is coming from 192 = 4 * 48;
If you say so we can shrink the buffer size, when I've looked at the code 16K was the safe upper bound I've calculated, it is possible that I've overlooked something.

Reply:
I still don't understand. How about this: 
```suggestion
// Mixxx engine uses audio buffers up to ~80 ms. This is 8192 Frames @ 96 kHz
// see class SoundManagerConfig::AudioBufferSizeIndex
// 96 kHz * 80 ms = 7680 -> 8192 (2^13)
static constexpr unsigned int kMaxEngineFrames = 8192;
```

Than we need to have a close look into the scalers that have t o deal with 192kHz
Maybe in separate PR? 
Keep the original here:  
`constexpr unsigned int MAX_BUFFER_LEN = 160000;`
where we are unsure?

Reply:
I'm missing something.
It seems that your suggestions is for 96kHz, or it seems to me that the apps can run at up to 192kHz. So it'd double up the max number of frames per process call right?